src/core/event.py

```diff
@@ -0,0 +1,72 @@
+from typing import Tuple
+from core.integrator.integrator import propagate_state
+from core.models.model_list import ModelContainer
+from core.state.state import State, ObservedState
+from core.state.statetime import StateTime
+
+
+class Event:
+    """Representation of a sim event with a list of models to be evaluated"""
+
+    def __init__(self, model_container: ModelContainer):
+        self.model_container = model_container
+
+    def evaluate(self, state_time: StateTime) -> Tuple[StateTime, ObservedState]:
+        """Evaluates all the models for this event
+
+        Args:
+            state_time (StateTime): The StateTime object at which this event takes place
+
+        Returns:
+            Tuple[StateTime, ObservedState]: The propagated statetime and observed state
+        """
+        ...
+
+
+class NormalEvent(Event):
+    """Representation of a normal event"""
+
+    def __init__(self, model_container: ModelContainer):
+        super().__init__(model_container)
+
+    def evaluate(self, state_time: StateTime) -> Tuple[StateTime, ObservedState]:
+        """Evaluates all the models for this event
```

Comment:
A side effect of this method would be that state_time is updated right? Maybe this should be written in the spec, or maybe we shouldn't be changing it. I know in terms of the current usage, the "state_time" is overwritten using this call, but as a way to make the code more decoupled we should probably be more careful here.

Reply:
Resolved offline. This will be in a separate PR.